File: examples/c++/button_delete.cpp
class ButtonDelete {
    DataTable& table_;
public:
    ButtonDelete(DataTable& table) : table_(table) {}
    void onClick(int rowId) {
        table_.removeRecord(rowId);
        if (table_.selectedRowId == rowId) table_.selectedRowId = -1;
    }
    void confirm() {
        if (table_.selectedRowId >= 0) onClick(table_.selectedRowId);
    }
};

File: examples/python/button_delete.py
class ButtonDelete:
    def __init__(self, table):
        self.table = table
        self.selected_row_id = -1

    def on_click(self, row_id):
        self.table.remove_record(row_id)
        if self.selected_row_id == row_id:
            self.selected_row_id = -1

    def confirm(self):
        if self.selected_row_id >= 0:
            self.on_click(self.selected_row_id)

    def render(self):
        return ''
